### PR TITLE
feat(popover):add popover margin prop

### PR DESCRIPTION
**Pull request by jiaoy14**
`Fixes #1, Fixes #2`

### CODE REVIEW

**Review by chasestarr — COMMENTED** [review]
Can you please add a VRT validating the feature?

**Review by sandgraham — COMMENTED** [review]
We should add this prop to the sandbox on the doc site. You can do that in:
`baseui/documentation-site/components/yard/config/popover.ts`

**Review by chasestarr — APPROVED** [review]
Awesome! this lgtm

File: src/popover/types.js [hunk 1]
```diff
@@ -109,6 +109,8 @@ export type BasePopoverPropsT = {
   showArrow?: boolean,
   /** Whether to toggle the popover when trigger is clicked or hovered. */
   triggerType: TriggerTypeT,
+  /** Margin of the popover */
+  popoverMargin: number,
```
Comment (chasestarr) [inline src/popover/types.js:113]: Sorry I missed this, we need to make the prop optional because it will fail flow check for existing applications

**Review by chasestarr — CHANGES_REQUESTED** [review]

**Review by chasestarr — APPROVED** [review]
thanks for the type fix

**Pull request merged**